metric_learn/covariance.py

```diff
@@ -34,3 +34,8 @@ def fit(self, X, y=None):
     else:
       self.M_ = np.linalg.inv(self.M_)
     return self
+
+class Covariance(_Covariance, UnsupervisedMixin):
```

Comment:
If nothing else will use `_Covariance`, I'd prefer to eliminate it and use the `UnsupervisedMixin` directly.

Reply:
In general, I don't think it's worth separating out the algorithm implementation unless/until we have >1 class that needs it.